src/index.ts

```diff
@@ -62,50 +118,109 @@ export default class Account {
    * ~~~
    *
    * @param data
-   * An account can be initialized with either a `buffer` containing the RLP serialized account.
-   * Or an `Array` of buffers relating to each of the account Properties, listed in order below.
-   *
+   * An account can be initialized with either:
+   * 1. A `buffer` containing the RLP serialized account.
+   * 2. An `Array` of buffers relating to each of the account properties listed in order [nonce, balance, stateRoot, codeHash].
+   * 3. An `object` with properties {nonce, balance, stateRoot, codeHash}.
    * For `Object` and `Array` each of the elements can either be a `Buffer`, hex `String`, `Number`, or an object with a `toBuffer` method such as `Bignum`.
    */
   constructor(data?: any) {
-    const fields = [
-      {
-        name: 'nonce',
-        default: Buffer.alloc(0),
-      },
-      {
-        name: 'balance',
-        default: Buffer.alloc(0),
-      },
-      {
-        name: 'stateRoot',
-        length: 32,
-        default: ethUtil.KECCAK256_RLP,
-      },
-      {
-        name: 'codeHash',
-        length: 32,
-        default: ethUtil.KECCAK256_NULL,
-      },
-    ]
-
-    ethUtil.defineProperties(this, fields, data)
+    let nonce
+    let balance
+    let stateRoot
+    let codeHash
+
+    if (typeof data === 'string') {
+      data = Buffer.from(data.substring(0, 2) === '0x' ? data.substring(2) : data, 'hex')
```

Comment:
We normally use `toBuffer` for this kind of conversions, which has some validation logic. For example, it will throw if a string without the `0x`-prefix is passed.